kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -203,11 +203,34 @@ public CloseableIterator<FilteredColumnarBatch> getAddFilesAsColumnarBatches(
   protected Tuple2<Protocol, Metadata> loadTableProtocolAndMetadata(
       Engine engine, Optional<SnapshotHint> snapshotHint, long snapshotVersion) {
 
-    // Exit early if the hint already has the info we need
+    // If the snapshot hint is provided and matches the expected version, load the P&M from version hint.
+    // If not, e.g. for the case of time travel, skip use version hint and fallback to CRC file or log replay.
     if (snapshotHint.isPresent() && snapshotHint.get().getVersion() == snapshotVersion) {
       return new Tuple2<>(snapshotHint.get().getProtocol(), snapshotHint.get().getMetadata());
     }
 
+    // Compute the lower bound for the CRC search
+    // If the snapshot hint is present, we can use it as the lower bound for the CRC search.
+    // TODO: this can be further improved to make the lower bound until the checkpoint version
+    Optional<Long> crcSearchLowerBound = snapshotHint.map(SnapshotHint::getVersion);
```

Comment:
This doesn't seem right to me. If the target version is 13 and the snapshot hint is at version 16, then we are setting the crcSearchLowerBound to 16?

What was wrong with my suggestion of checking in the constructor of LogReplay if the snapshotHint is greater than the target version, and just setting the snapshotHint to empty?